phantom_token.c

```diff
@@ -811,3 +808,70 @@ static char* set_client_credential_configuration_slot(ngx_conf_t *config_setting
 
     return "invalid_client_credential";
 }
+
+/*
+ * Add the error response as a JSON object that is easier to handle than the default HTML response that NGINX returns
+ * Note that for SPAs, the OAuth Proxy plugin runs before this one and adds CORS headers so that the browser can read this payload
+ * http://nginx.org/en/docs/dev/development_guide.html#http_response_body
+ */
+static ngx_int_t write_error_response(ngx_http_request_t *request, ngx_int_t status, phantom_token_configuration_t *module_location_config)
+{
+    ngx_int_t rc;
+    ngx_str_t code;
+    ngx_str_t message;
+    u_char jsonErrorData[256];
+    ngx_chain_t output;
+    ngx_buf_t *body = NULL;
+    const char *errorFormat = NULL;
+    size_t errorLen = 0;
+
+    if (request->method != NGX_HTTP_HEAD)
+    {
+        body = ngx_calloc_buf(request->pool);
+        if (body == NULL)
+        {
+            ngx_log_error(NGX_LOG_WARN, request->connection->log, 0, "Failed to allocate memory for error body");
+            return NGX_HTTP_INTERNAL_SERVER_ERROR;
+        }
+        else
+        {
+            if (status == NGX_HTTP_UNAUTHORIZED)
+            {
+                ngx_str_set(&code, "unauthorized_request");
+                ngx_str_set(&message, "Access denied due to missing, invalid or expired credentials");
+            }
+            else
+            {
+                ngx_str_set(&code, "server_error");
+                ngx_str_set(&message, "Problem encountered processing the request");
+            }
+
+            /* The string length calculation replaces the two '%V' markers with their actual values */
+            errorFormat = "{\"code\":\"%V\", \"message\":\"%V\"}";
```

Comment:
I suggest removing the space after the comma

Reply:
done